docs/cloud/execution/custom_environment.md

```diff
@@ -0,0 +1,128 @@
+# Custom Environment
+
+[[toc]]
+
+Prefect Environments allow for completely custom user created Environments. The only requirements is that your custom environment inherits from the base `Environment` class.
```

Comment:
Actually, you should include a warning or just a call out that custom environments should also 
have a different name than any of Prefect's environment classes.  (This probably won't break anything but could result in weird behavior considering our serializers work by inspecting class name)